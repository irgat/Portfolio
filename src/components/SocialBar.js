import React from 'react';
import PropTypes from 'prop-types';
import Icon from './Icon';

const propTypes = {
    items: PropTypes.array.isRequired,
    className: PropTypes.string,
};

const defaultProps = {
    className: undefined,
};

class SocialBar extends React.Component {
    filterValidItems = (itemsList) => {
        return itemsList.filter(item => item.imagePath !== undefined);
    };

    render() {
        let validItems = this.filterValidItems(this.props.items);
        return (
            (validItems.length > 0) && <ul className={this.props.className}>
                {validItems.map((item, id) => (
                    <Icon
                        key={id}
                        {...item}
                    />
                ))}
            </ul>
        );
    }
}

SocialBar.propTypes = propTypes;
SocialBar.defaultProps = defaultProps;

export default SocialBar;
