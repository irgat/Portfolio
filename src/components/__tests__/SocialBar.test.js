import React from 'react';
import Enzyme from 'enzyme';
import SocialBar from '../SocialBar';

const invalidList = [
    {},
    { key: 'value' },
    { url: 'https://www.w3.org' },
    { url: 'https://www.w3.org', target: '_blank' },
    { target: '_blank' },
];

const mixedList = [
    { imagePath: 'folderName/fileName.ext' },
    { imagePath: 'folderName/fileName.ext', url: 'https://www.w3.org' },
    { imagePath: 'folderName/fileName.ext', url: 'https://www.w3.org', target: '_blank' },
    { imagePath: 'folderName/fileName.ext', target: '_blank' },
].concat(invalidList);

const socialBar = (itemList) => {
    return Enzyme.shallow(<SocialBar items={itemList} />);
};

describe('<SocialBar />', () => {
    it('filterValidItems() returns 4 items', () => {
        expect(socialBar(mixedList).instance().filterValidItems(mixedList)).toMatchSnapshot();
    });

    it('returns null when there is no valid items', () => {
        expect(socialBar(invalidList).find('ul').getElements()).toMatchSnapshot();
    });

    describe('<ul />', () => {
        const wrapper = socialBar(mixedList);
        const validItems = wrapper.instance().filterValidItems(mixedList);
        it(`receives ${mixedList.length} item(s), renders ${validItems.length} valid <Icon />(s)`, () => {
            expect(wrapper.find('Icon')).toHaveLength(validItems.length);
            expect(wrapper.find('ul').getElements()).toMatchSnapshot();
        });
    });
});
