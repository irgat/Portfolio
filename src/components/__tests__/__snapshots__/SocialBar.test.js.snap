// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialBar /> <ul /> receives 9 item(s), renders 4 valid <Icon />(s) 1`] = `
Array [
  <ul>
    <Icon
      imagePath="folderName/fileName.ext"
      target="_self"
      url="#"
    />
    <Icon
      imagePath="folderName/fileName.ext"
      target="_self"
      url="https://www.w3.org"
    />
    <Icon
      imagePath="folderName/fileName.ext"
      target="_blank"
      url="https://www.w3.org"
    />
    <Icon
      imagePath="folderName/fileName.ext"
      target="_blank"
      url="#"
    />
  </ul>,
]
`;

exports[`<SocialBar /> filterValidItems() returns 4 items 1`] = `
Array [
  Object {
    "imagePath": "folderName/fileName.ext",
  },
  Object {
    "imagePath": "folderName/fileName.ext",
    "url": "https://www.w3.org",
  },
  Object {
    "imagePath": "folderName/fileName.ext",
    "target": "_blank",
    "url": "https://www.w3.org",
  },
  Object {
    "imagePath": "folderName/fileName.ext",
    "target": "_blank",
  },
]
`;

exports[`<SocialBar /> returns null when there is no valid items 1`] = `Array []`;
